tests/src/configresolver.test.cpp

```diff
@@ -0,0 +1,35 @@
+// Tests for configuration resolver inheritance and ambiguity.
+#include "doctest.h"
+#include "luteprojectroot.h"
+
+#include "lute/configresolver.h"
+
+#include "Luau/FileUtils.h"
+#include "Luau/LuauConfig.h"
```

Comment:
```suggestion
#include "Luau/Config.h"
#include "Luau/FileUtils.h"
```
I don't think we need LuauConfig.h here?

Reply:
LuauConfig.h is specifically for `.config.luau` file extraction. Config.h is the one for `.luaurc` files and anywhere you see `Luau::Config` in the code.

Reply:
the configresolver is handling both `.luaurc` and `.config.luau` files tho so I think I do need `LuauConfig.h` for `Luau::kConfigName`

Reply:
We're not using anything from that in this file though. (You're right that we need it in `configresolver.cpp`, but I don't think we need it anywhere else..?)